automated/linux/vdsotest/vdsotest.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+. ../../lib/sh-test-lib
+OUTPUT="$(pwd)/output"
+RESULT_FILE="${OUTPUT}/result.txt"
+RESULT_LOG="${OUTPUT}/result_log.txt"
+TMP_LOG="${OUTPUT}/tmp_log.txt"
+TEST_PASS_LOG="${OUTPUT}/test_pass_log.txt"
+TEST_FAIL_LOG="${OUTPUT}/test_fail_log.txt"
+TEST_SKIP_LOG="${OUTPUT}/test_skip_log.txt"
+TEST_METRIC_LOG="${OUTPUT}/test_metric_log.txt"
+METRIC_FILE="${OUTPUT}/metric.txt"
+
+# set it to VDSO_INSTALL_PATH=/opt/vdsotest if you want to use git
+VDSO_INSTALL_PATH=/usr
+TEST_PROGRAM=vdsotest
+TEST_PROG_VERSION=
+TEST_GIT_URL=https://github.com/nathanlynch/vdsotest.git
+TEST_DIR="$(pwd)/${TEST_PROGRAM}"
+SKIP_INSTALL="false"
+API=""
+DURATION=""
+VDSOTESTALL="yes"
+TEST_TYPE=""
+usage() {
+	echo "\
+	Usage: [sudo] ./vdso.sh [-a <API>]
+				[-d <DURATION>]
+				[-f <ALL>]
+				[-t <TEST-TYPE>]
+				[-v <TEST_PROG_VERSION>]
+				[-u <TEST_GIT_URL>]
+				[-p <TEST_DIR>]
+				[-s <true|false>]
+				[-h help]
+
+	<API>:
+	where API must be one of:
+	clock-gettime-monotonic
+	clock-getres-monotonic
+	clock-gettime-monotonic-coarse
+	clock-getres-monotonic-coarse
+	clock-gettime-monotonic-raw
+	clock-getres-monotonic-raw
+	clock-gettime-tai
+	clock-getres-tai
+	clock-gettime-boottime
+	clock-getres-boottime
+	clock-gettime-realtime
+	clock-getres-realtime
+	clock-gettime-realtime-coarse
+	clock-getres-realtime-coarse
+	getcpu
+	gettimeofday
+
+	<DURATION>:
+	Time in long will the test be running. DURATION can be set
+	to X
+	default: 1s - seconds
+
+	<ALL>:
+	Run all tests
+	default: all
+
+	<TEST_TYPE>:
+	TEST_TYPE must be one of:
+	verify
+	bench
+	abi
+
+	<TEST_PROG_VERSION>:
+	If this parameter is set, then the ${TEST_PROGRAM} is cloned. In
+	particular, the version of the suite is set to the commit
+	pointed to by the parameter. A simple choice for the value of
+	the parameter is, e.g., HEAD. If, instead, the parameter is
+	not set, then the suite present in TEST_DIR is used.
+
+	<TEST_GIT_URL>:
+	If this parameter is set, then the ${TEST_PROGRAM} is cloned
+	from the URL in TEST_GIT_URL. Otherwise it is cloned from the
+	standard repository for the suite. Note that cloning is done
+	only if TEST_PROG_VERSION is not empty
+
+	<TEST_DIR>:
+	If this parameter is set, then the ${TEST_PROGRAM} suite is cloned to or
+	looked for in TEST_DIR. Otherwise it is cloned to /opt/${TEST_PROGRAM}
+
+	If next parameter is set, then the vdso suite is cloned to or
+	looked for in VDSO_INSTALL_PATH. Otherwise it is cloned to /opt/vdso
+	<VDSO_INSTALL_PATH>
+
+	<SKIP_INSTALL>:
+	If you already have it installed into the rootfs.
+	default: false"
+}
+
+while getopts "a:d:f:i:t:h:p:u:s:v:" opt; do
```

Comment:
When I tried to view the help I saw this
$ ./vdsotest.sh -h                               
./vdsotest.sh: option requires an argument -- h

Change "...:h:p:..." to "...:hp:..."